### ISSUE TITLE
Cannot get logged/authenticated in contact id

### ISSUE THREAD

**Issue by davidjosephhayes** [mecachisenros/civicrm-wp-rest#8]
```
$contact_id = CRM_Core_Session::singleton()->getLoggedInContactID(); 
````

`$contact_id` is null when using this plugin, but not when calling `rest.php`. I am not sure if this is expected behavior or not. I see quite a few mentions of session across here: [https://lab.civicrm.org/dev/wordpress/issues/20#civicrm-rest-api](https://lab.civicrm.org/dev/wordpress/issues/20#civicrm-rest-api), but I do not know how that relates to CiviCRM and Wordpress.

**Comment by mecachisenros** [comment]
@davidjosephhayes can you give an example as into what context `getLoggedInContactID()` is `null`?

**Comment by davidjosephhayes** [comment]
As far as I can tell it is all of them.

One place I noticed it was when a contact just had permission to view their own contact but not all contacts. A request with that contacts API key returned no contacts. I also saw the behavior on an wrapper around another API I had from a custom extension that was limiting some results by contact id.

Here is a stock instance of Wordpress 5.3.2 and CiviCRM 5.21.2. The api key here belongs to a user with a Subscriber role and standard permissions except for "view my contact". The request will return no values on first request, then 1 value on second request (due to a wordpress_logged_in cookie being set and logging you into Wordpress as the user associated with the API key).

[https://one.blackbricksoftware.com/wp-json/civicrm/v3/rest?entity=Contact&action=get&key=647a6815e8ac805fa00c6327f097a974&api_key=02LA7NmURmm99fb9aTTElcGp&json=1](https://one.blackbricksoftware.com/wp-json/civicrm/v3/rest?entity=Contact&action=get&key=647a6815e8ac805fa00c6327f097a974&api_key=02LA7NmURmm99fb9aTTElcGp&json=1)
